ui/app/pages/permissions-connect/permissions-connect.container.js

```diff
@@ -33,6 +42,18 @@ const mapStateToProps = (state, ownProps) => {
     addressLastConnectedMap[key] = formatDate(addressLastConnectedMap[key], 'yyyy-M-d')
   })
 
+  const connectPath = `${CONNECT_ROUTE}/${permissionsRequestId}`
+  const confirmPermissionPath = `${CONNECT_ROUTE}/${permissionsRequestId}${CONNECT_CONFIRM_PERMISSIONS_ROUTE}`
+
+  let page = ''
+  if (pathname === connectPath) {
+    page = '1'
+  } else if (pathname === confirmPermissionPath) {
+    page = '2'
+  } else {
+    history.push(DEFAULT_ROUTE)
```

Comment:
When I suggested a redirect as a possible solution, I was thinking of the React Router component `Redirect`, though of course this is a `mapStateToProps` so that doesn't make sense 🤔 

The problem with this is that the component is still going to render with a blank "page" set, so the permissions header will say "/2" for a moment while the page loads, which is not ideal. I think throwing an error would be a better way to handle this case, as it should never happen anyway unless the user directly manipulates the address bar or something.